### DISCUSSION TITLE
edit default api

### DISCUSSION THREAD

**Discussion by sthisfake** [pocketbase/pocketbase#2788] [Q&A] [answered]
hey there , i have a question , how can i change the behaviour of a default api 
for example i have a "test" collection with some json fields , i want to store the data in some custom way and add some validation and custom error exceptions to it . for example the default api address is in this /api/collections/test/records that pocketbase autogenerates.

**Comment by ganigeorgiev** [comment, marked as answer]
Currently the only way to modify the default Record APIs behavior is to use PocketBase as Go framework and register to the related [Record API hooks](https://pocketbase.io/docs/event-hooks/#record-crud-api-hooks).

With the upcoming v0.17.0 you'll also have an option to use JavaScript (_you can find some details in https://github.com/pocketbase/pocketbase/discussions/2767#discussioncomment-6245393_).
